packages/xchain-ethereum/src/client.ts

```diff
@@ -626,20 +612,26 @@ export default class Client extends BaseXChainClient implements XChainClient, Et
    * @returns {BigNumber} The estimated gas limit.
    */
   estimateApprove = async ({
-    spender,
-    sender,
+    contractAddress,
+    spenderAddress,
     amount,
-  }: Omit<ApproveParams, 'feeOptionKey' | 'walletIndex'>): Promise<BigNumber> => {
+    gasLimitFallback,
+    walletIndex,
+  }: Omit<ApproveParams, 'feeOptionKey'>): Promise<BigNumber> => {
     try {
       const txAmount = amount ? BigNumber.from(amount.amount().toFixed()) : MAX_APPROVAL
-      const gasLimit = await this.estimateCall(sender, erc20ABI, 'approve', [
-        spender,
-        txAmount,
-        { from: this.getAddress() },
-      ])
+      const gasLimit = await this.estimateCall({
+        walletIndex,
+        contractAddress,
+        abi: erc20ABI,
+        funcName: 'approve',
+        funcParams: [spenderAddress, txAmount, { from: this.getAddress() }],
+      })
 
       return gasLimit
     } catch (error) {
+      // return `gasLimitFallback` if available
+      if (gasLimitFallback) return BigNumber.from(gasLimitFallback)
```

Comment:
if the caller has to provide the gas limitFallback as a param (only to treturn it back),  can we just omit it altogether  and let the caller implement their own fallback logic outside of this call?

example

```typescript
let gasLimit: BigNumber
try {
    gaslimit = ethClient.estimateApprove({args})
}catch (error){
    console.log('oopsie cannot calc gaslimit!')
}
if(!gasLimit){
    gasLimit = fallbackGasLimit
}
```

Reply:
> let the caller implement their own fallback logic outside of this call
Agree. I'm going to remove `limitFallback` param for `estimateApprove`